CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -161,8 +166,8 @@ local function CreateTopBar()
 	end
 
 	function this:SetTopbarDisplayMode(opaque)
-		topbarContainer.BackgroundTransparency = opaque and TOPBAR_OPAQUE_TRANSPARENCY or TOPBAR_TRANSLUCENT_TRANSPARENCY
-		topbarShadow.Visible = not opaque
+		topbarContainer.BackgroundTransparency = (not topbarEnabled and 1) or (opaque and TOPBAR_OPAQUE_TRANSPARENCY) or TOPBAR_TRANSLUCENT_TRANSPARENCY
```

Comment:
This line and the above BackgroundTransparency lines are getting out of hand.
I suggest we create an extra function in Topbar called ComputeTransparency() to do all this calculations in an easy to read way.